sysprep/activate_instance.ps1

```diff
@@ -266,19 +266,36 @@ function Verify-ActivationStatus {
   [String]$activation_status = $null
   [String]$status = $null
 
-  try {
-    $slmgr_status = & cscript //E:VBScript //nologo $env:windir\system32\slmgr.vbs /dli
-  }
-  catch {
+  # Server 2008 doesn't store activation status in registry; check slmgr
+  if([Environment]::OSVersion.Version.Major -eq 6 -and [Environment]::OSVersion.Version.Minor -le 1)
+  {
+    try {
+      $slmgr_status = & cscript //E:VBScript //nologo $env:windir\system32\slmgr.vbs /dli
+    }
+    catch {
+      return $active
+    }
+    $status = $slmgr_status | Select-String -Pattern '^License Status:'
+    # The initial space is to ensure "Unlicensed" does not match.
+    if ($status -match ' Licensed') {
+      $active = $true
+    }
     return $active
```

Comment:
Instead of having the return in both the if and else. It should be after the if/else block so it's a single return that always happens.

Reply:
Good point; done.